File: java/src/main/java/algorithm/FindF.java
package algorithm;

import java.io.IOException;
import java.util.List;

public class FindF {

    private final Helper helper = new Helper();
    private final String fn;

    public FindF(String fn) {
        this.fn = fn;
    }

    public String find(OneTwo ot) throws IOException {
        List<String> data = helper.l(fn);
        List<F> fData = p(data);
        F r = helper.find(fData, ot);
        return r.i();
    }

    List<F> p(List<String> data) {
        return helper.p(data, //
                this::startsWithRank, //
                this::newF);
    }

    private boolean startsWithRank(String s) {
        return s.matches("\\s*\\d+\\.\\s.*");
    }

    private F newF(String[] c) {
        int cTeam = 1;
        int cF = 6;
        int cA = 8;

        String team = c[cTeam];
        int f = Integer.parseInt(c[cF]);
        int a = Integer.parseInt(c[cA]);

        return new F(team, f, a);
    }

}

File: javascript/src/find-f.js
const { Helper } = require('./helper.js');
const { F } = require('./f.js');

class FindF {
    constructor(fn) {
        this.helper = new Helper();
        this.fn = fn;
    }

    find(ot) {
        const data = this.helper.l(this.fn);
        const fData = this.p(data);
        const f = this.helper.find(fData, ot);
        return f.r();
    }

    p(data) {
        return this.helper.p(data, this.startsWithRank, this.newF);
    }

    startsWithRank(s) {
        return /^\s*\d+\.\s.*/u.test(s);
    }

    newF(c) {
        const cTeam = 1;
        const cF = 6;
        const cA = 8;

        const team = c[cTeam];
        const f = parseInt(c[cF], 10);
        const a = parseInt(c[cA], 10);

        return new F(team, f, a);
    }
}

module.exports = { FindF };
